data_structures/src/transaction_factory.rs

```diff
@@ -202,19 +219,21 @@ pub fn build_commit_collateral(
 /// Generic inputs/outputs builder: can be used to build
 /// value transfer transactions and data request transactions.
 #[allow(clippy::too_many_arguments)]
-fn build_inputs_outputs_inner(
+pub fn build_inputs_outputs_inner<T>(
     outputs: Vec<ValueTransferOutput>,
     dr_output: Option<&DataRequestOutput>,
     fee: u64,
-    own_utxos: &mut OwnUnspentOutputsPool,
+    utxos: &mut T,
     own_pkh: PublicKeyHash,
-    all_utxos: &UnspentOutputsPool,
     timestamp: u64,
     tx_pending_timeout: u64,
     // The block number must be lower than this limit
     block_number_limit: Option<u32>,
     utxo_strategy: UtxoSelectionStrategy,
-) -> Result<(Vec<Input>, Vec<ValueTransferOutput>), TransactionError> {
+) -> Result<(Vec<Input>, Vec<ValueTransferOutput>), TransactionError>
+where
+    T: UtxoFantasy,
+{
     // On error just assume the value is u64::max_value(), hoping that it is
     // impossible to pay for this transaction
     let output_value: u64 = transaction_outputs_sum(&outputs)
```

Comment:
I think this was past me being lazy, can you rewrite this and instead of using `saturating_add`, return `OutputValueOverflow` on overflow?